Refactor ForeachUnaryOp.cu

izdeby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52450 wip
* #49039 [wip] Replace optimizers in torch.optim with the ones from torch.optim._multi_tensor
* #52449 Add type promotion to a fast path for foreach APIs
* #52448 Support broadcasting via slow path
* #48224 [wip] Pushed unsupported scenarios to slow path
* #51170 Updated foreach pointwise ops to support complex and bloat16 on CUDA
* #49714 Update foreach max/min
* #51060 Refactor foreach pointwise ops tests to use OpInfo
* #51059 Refactor foreach binary ops tests with tensor lists to use OpInfo
* #51058 Refactor foreach binary ops tests with scalars to use OpInfo
* #49712 Refactor foreach unary ops tests to use OpInfo
* #49250 Update foreach binary ops with a single scalar and list of tensors
* #49249 Update foreach binary ops with a scalar list
* #51131 Refactor ForeachUtils.h
* **#51894 Refactor ForeachUnaryOp.cu**
* #51893 Update foreach APIs to use scalar lists

original PR - https://github.com/pytorch/pytorch/pull/49248

Differential Revision: [D26323605](https://our.internmc.facebook.com/intern/diff/D26323605)

-------
Refactor ForeachUnaryOps.cu to better see what dtypes are supported for ops.
Update abs test to test ints.

izdeby committed:
Refactor ForeachUnaryOp.cu

[ghstack-poisoned]

Commit:
Update on "Refactor ForeachUnaryOp.cu"


original PR - https://github.com/pytorch/pytorch/pull/49248

Differential Revision: [D26323605](https://our.internmc.facebook.com/intern/diff/D26323605)

[ghstack-poisoned]

Commit:
Update on "Refactor ForeachUnaryOp.cu"


original PR - https://github.com/pytorch/pytorch/pull/49248

Differential Revision: [D26323605](https://our.internmc.facebook.com/intern/diff/D26323605)

[ghstack-poisoned]

izdeby commented:
@ngimel, @zou3519, resubmitting this PR. [Original one](https://github.com/pytorch/pytorch/pull/49248) was reverted due to failures.

ngimel commented:
Did you fix the failures? Note, 11.2 config in CI doesn't run optim tests at all because of cuda 11.2 bug (or mta bug, we are still investigating).

Commit:
Update on "Refactor ForeachUnaryOp.cu"


original PR - https://github.com/pytorch/pytorch/pull/49248

Differential Revision: [D26323605](https://our.internmc.facebook.com/intern/diff/D26323605)

-------
Refactor ForeachUnaryOps.cu to better see what dtypes are supported for ops.
Update abs test to test ints.

[ghstack-poisoned]

Commit:
Update on "Refactor ForeachUnaryOp.cu"


original PR - https://github.com/pytorch/pytorch/pull/49248

Differential Revision: [D26323605](https://our.internmc.facebook.com/intern/diff/D26323605)

-------
Refactor ForeachUnaryOps.cu to better see what dtypes are supported for ops.
Update abs test to test ints.

[ghstack-poisoned]

Commit:
Update on "Refactor ForeachUnaryOp.cu"


original PR - https://github.com/pytorch/pytorch/pull/49248

Differential Revision: [D26323605](https://our.internmc.facebook.com/intern/diff/D26323605)

-------
Refactor ForeachUnaryOps.cu to better see what dtypes are supported for ops.
Update abs test to test ints.

[ghstack-poisoned]

Commit:
Update on "Refactor ForeachUnaryOp.cu"


original PR - https://github.com/pytorch/pytorch/pull/49248

Differential Revision: [D26323605](https://our.internmc.facebook.com/intern/diff/D26323605)

-------
Refactor ForeachUnaryOps.cu to better see what dtypes are supported for ops.
Update abs test to test ints.

[ghstack-poisoned]

Commit:
Update on "Refactor ForeachUnaryOp.cu"


original PR - https://github.com/pytorch/pytorch/pull/49248

Differential Revision: [D26323605](https://our.internmc.facebook.com/intern/diff/D26323605)

-------
Refactor ForeachUnaryOps.cu to better see what dtypes are supported for ops.
Update abs test to test ints.

[ghstack-poisoned]

Commit:
Update on "Refactor ForeachUnaryOp.cu"


original PR - https://github.com/pytorch/pytorch/pull/49248

Differential Revision: [D26323605](https://our.internmc.facebook.com/intern/diff/D26323605)

-------
Refactor ForeachUnaryOps.cu to better see what dtypes are supported for ops.
Update abs test to test ints.

[ghstack-poisoned]
